hub/public-endpoints.mdx

```diff
@@ -0,0 +1,249 @@
+}
+```
+
+**Parameters:**
+- `prompt` (string, required): Text description of the desired image.
+- `width` (integer, default: 1024): Image width in pixels (64-2048).
+- `height` (integer, default: 1024): Image height in pixels (64-2048).
+- `num_inference_steps` (integer, default: 4): Number of denoising steps (1-8).
+- `guidance` (float, default: 1.0): Prompt adherence strength (0.5-2.0).
+- `seed` (integer, optional): Random seed for reproducible results.
+
+<Warning>
+Flux Schnell is optimized for speed and works best with lower step counts. Using higher values may not improve quality significantly.
+</Warning>
+
+## Python example
+
+Here is an example Python API request to Flux Dev using the `/run` endpoint:
+
+```python
+import requests
+
+headers = {
+    'Content-Type': 'application/json',
+    'Authorization': 'Bearer RUNPOD_API_KEY'
+}
+
+data = {
+    'input': {"prompt":"A serene mountain landscape at sunset","image_format":"png","num_inference_steps":25,"guidance":7,"seed":50,"width":1024,"height":1024}
+}
+
+response = requests.post('https://api.runpod.ai/v2/black-forest-labs-flux-1-dev/run', headers=headers, json=data)
+```
+
+You can generate public endpoint API requests for Python and other programming languages using the [public endpoint playground](#public-endpoint-playground).
+
+## Pricing
+
+Public endpoints use transparent, usage-based pricing:
+
+| Model | Price | Billing unit |
+|-------|-------|--------------|
+| Flux Dev | $0.02 | Per megapixel |
+| Flux Schnell | $0.00024 | Per megapixel |
+
+**Pricing examples:**
+- 512×512 image (0.25 megapixels): \$0.005 (Flux Dev) / \$0.00006 (Flux Schnell)
+- 1024×1024 image (1 megapixel): \$0.02 (Flux Dev) / \$0.00024 (Flux Schnell)
+- 2048×2048 image (4 megapixels): \$0.08 (Flux Dev) / \$0.00096 (Flux Schnell)
+
+<Note>
+Pricing is calculated based on the actual output resolution. You will not be charged for failed generations.
+</Note>
+
+## Best practices
+
+When working with public endpoints, following best practices will help you achieve better results and optimize performance. 
+
+### Prompt engineering
+
+For prompt engineering, be specific with detailed prompts as they generally produce better results. Include style modifiers such as art styles, camera angles, or lighting conditions. For Flux Dev, use negative prompts to exclude unwanted elements from your images.
+
+A good prompt example would be: "A professional portrait of a woman in business attire, studio lighting, high quality, detailed, corporate headshot style."
+
+### Performance optimization
+
+For performance optimization, choose the right model for your needs. Use Flux Schnell when you need speed, and Flux Dev when you need higher quality. Standard dimensions like 1024×1024 render fastest, so stick to these unless you need specific aspect ratios. For multiple images, use asynchronous endpoints to batch your requests. Consider caching results by storing generated images to avoid regenerating identical prompts.
```

Comment:
Do we need to note that when using the playground we bin the aspect ratio for the user? Or is that TMI?